### PR TITLE
Clean up style-guide.scss

### LINKED ISSUE #6396: Improve style-guide.scss
We can simplify [this huge chunk of repetitive code](https://github.com/mozilla/foundation.mozilla.org/blob/master/source/sass/views/style-guide.scss#L8-L98) by creating a colour map and use [SCSS's `each` function](https://sass-lang.com/documentation/at-rules/control/each#with-maps) to create the corresponding class & ruleset.

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #6396 

@danielfmiranda This is a pretty straight-forward code cleanup PR. Assigning you as the reviewer so you can familiarize yourself with the PR review flow & SCSS.

These are steps I would go through if I was the reviewer:

1. Check if there's any REAL visual regression issue flagged by Percy. If yes, that's a bad sign 😂 (Note Percy sometimes has false-positive flags so the result can't always be trusted 😬 )
2. Check where those color classes are being used in the codebase. For example, search for `gray-60` in all `.html` files.
3. Turns out they are mainly used in `styleguide.html`. (http://test.example.com:8000/en/style-guide/)
4. Go to Percy and verify "Styleguide page" looks as before.
5. Verify code looks good.
6. Leave PR review. Guide: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews

### CODE REVIEW

File: source/sass/mofo-bootstrap/_colors.scss [hunk 1]
```diff
@@ -44,3 +44,31 @@ $youtube-regrets-light-red: #ffdcdc;
...
+  "festival-purple": $festival-purple,
+  "festival-gradient": $festival-gradient,
+  "pulse-pink": $pulse-pink,
+  "pulse-purple": $pulse-purple,
+  "youtube-regrets-red": $youtube-regrets-red,
+  "youtube-regrets-lighter-red": $youtube-regrets-lighter-red,
+  "youtube-regrets-light-red": $youtube-regrets-light-red,
+  "dear-internet-lilac": $dear-internet-lilac,
+  "dear-internet-pink": $dear-internet-pink,
+);
```
Comment (mmmavis) [inline source/sass/mofo-bootstrap/_colors.scss:74]: too bad this map has to be manually created

File: source/sass/buyers-guide/_colors.scss [hunk 2]
```diff
@@ -11,3 +11,18 @@ $pni-warning-yellow: #fbd545;
...
+  "pni-blue": $pni-blue,
+  "pni-blue-purple": $pni-blue-purple,
+  "pni-purple": $pni-purple,
+  "pni-purple-pink": $pni-purple-pink,
+  "pni-pink": $pni-pink,
+  "pni-gradient": $pni-gradient,
+  "pni-warning-yellow": $pni-warning-yellow,
+  "pni-product-list-background": $pni-product-list-background,
+  "pni-product-image-background": $pni-product-image-background,
+);
```
Comment (mmmavis) [inline source/sass/buyers-guide/_colors.scss:28]: too bad this map has to be manually created

**Review by danielfmiranda — APPROVED** [review]
Hi Mavis, thanks for writing up that guide!

I just looked over the code, as well as the changes in percy and verified that the style guide site remained the same.

Everything looks good to me, so I am approving!

**Comment by mmmavis** [comment]
@danielfmiranda woohooo! Thanks for the note!

**Pull request merged**